### DISCUSSION TITLE
wasm-bindgen for generic wasi interfaces?

### DISCUSSION THREAD

**Discussion by manuschillerdev** [wasm-bindgen/wasm-bindgen#2883] [Q&A] [answered]
In my understanding wasm-bindgen is developed specifcally for browser environments.
Is there something similar for generic typed wasi interfaces?

My specific usecase: I'd like to validate a json object (passed by reference) against an endpoint.
So I'd need to have a generic WASI request interface and a generic interface to pass the json object by reference.

I am just starting out with wasm/wasi so the question might be completely off for wasm-bindgen.. I'm happy for any correction or hints into the right direction how I could implement that case :) 

Thanks a lot for any input!! :)

**Comment by alexcrichton** [comment, marked as answer]
For WASI-like use cases I'd recommend looking into the [component model](https://github.com/webassembly/component-model) proposal for wasm.
